flux/client/http.go

```diff
@@ -46,40 +34,17 @@ func NewHTTP(u url.URL) (*HTTP, error) {
 }
 
 // Query runs a flux query against a influx server and decodes the result
-func (s *HTTP) Query(ctx context.Context, r *ProxyRequest) (flux.ResultIterator, error) {
-	qreq, err := QueryRequestFromProxyRequest(r)
-	if err != nil {
-		return nil, err
-	}
-	var body bytes.Buffer
-	if err := json.NewEncoder(&body).Encode(qreq); err != nil {
-		return nil, err
-	}
-
-	hreq, err := http.NewRequest("POST", s.url.String(), &body)
-	if err != nil {
-		return nil, err
-	}
+func (s *HTTP) Do(hreq *http.Request) (*http.Response, error) {
 	if s.Username != "" {
 		hreq.SetBasicAuth(s.Username, s.Password)
 	}
 
 	hreq.Header.Set("Content-Type", "application/json")
 	hreq.Header.Set("Accept", "text/csv")
-	hreq = hreq.WithContext(ctx)
+	// ?? hreq = hreq.WithContext(ctx)
```

Comment:
TODO: need to understand whether this client can be canceled / if it needs a cancelable context

Reply:
Figured it out I think - this is fine because we set the context on the request in flux: https://github.com/influxdata/flux/blob/master/stdlib/http/post.go#L73

Reply:
Though actually this appears to be for making a new http call, not the initial one to flux. So maybe it needs a re-look. Will make sure I can hit it in tests and understand what's up with it.